src/lib/mergeSchemas.js

```diff
@@ -77,4 +77,6 @@ export async function mergeSchemas() {
       },
     }),
   })
+  mergedSchema.__allowedLegacyNames = ["__id"]
```

Comment:
After tracing thru some of the `buildSchema`, etc. methods used above, this was the one that wound up working.

Which makes sense, the result of this function _is_ the schema that we are operating on, executing resolvers, etc. So this is the schema that needs this option set.

Reply:
@alloy should be able to comment more about this as he's done some recent relay work around this option